packages/main/src/plugin/kubernetes-client.ts

```diff
@@ -127,6 +179,26 @@ export class KubernetesClient {
     });
   }
 
+  setupKubeWatcher() {
+    this.kubeWatcher?.abort();
+    const ns = this.currrentNamespace;
+    if (ns) {
+      new Watch(this.kubeConfig)
+        .watch(
+          '/api/v1/namespaces/' + ns + '/pods',
+          {},
+          // eslint-disable-next-line @typescript-eslint/no-unused-vars
+          (phase: string, apiObj: any, watchObj?: any) => {
+            this.apiSender.send('pod-event');
+          },
+          (err: any) => {
+            console.log('Kube event error ' + err);
+          },
+        )
+        .then(req => (this.kubeWatcher = req));
```

Comment:
it seems we have a depth that is too big
if ()
   new 
      method
         then

probably need to call functions or use await to reduce the number of depth

Reply:
That's method chainging so I don't understand the issue here ?

Reply:
it's just about readability of the overall code. Lot of chaining does not mean it's readable.